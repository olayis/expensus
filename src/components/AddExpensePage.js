import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';
import { startChangeCurrency } from '../actions/locale';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.startChangeCurrency({ currency: 'Danish Krone' });
    this.props.history.push('/');
  };
  render() {
    return (
      <main className='main-content'>
        <div className='content-container'>
          <div className='add-expense-container'>
            <ExpenseForm onSubmit={this.onSubmit} />
          </div>
        </div>
      </main>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense)),
  startChangeCurrency: (currency) => dispatch(startChangeCurrency(currency)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
