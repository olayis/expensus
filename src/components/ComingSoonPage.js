import React from 'react';
import { Link } from 'react-router-dom';
import programming from '../../public/img/programming.svg';

const ComingSoonPage = () => (
  <main className='main-content'>
    <div className='content-container'>
      <div className='error'>
        <img
          src={programming}
          alt='Programmer'
          className='error__illustration'
          width='1075'
          height='585'
        />
        <h2 className='error__text'>This page is coming soon.</h2>
        <div className='error__explanation'>
          Continue managing your expenses.
          <span className='error__action'>
            <Link className='btn btn--try-again' to='/'>
              Go Home
            </Link>
          </span>
        </div>
      </div>
    </div>
  </main>
);

export default ComingSoonPage;
