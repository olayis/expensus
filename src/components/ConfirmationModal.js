import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => {
  if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

  return (
    <div>
      <Modal
        contentLabel='Are you sure?'
        isOpen={!!props.deleteSelected}
        onRequestClose={props.stopDeleteDataSelected}
        closeTimeoutMS={200}
        className='modal'
        aria={{
          labelledby: 'heading',
          describedby: 'full_description',
        }}
      >
        <h3 id='heading' className='modal__title'>
          Delete Data
        </h3>
        <p id='full_description' className='modal__body'>
          Are you sure you want to delete all of your data on Expensus?
        </p>
        <button className='modal__btn--delete' onClick={props.handleDeleteData}>
          Delete
        </button>
        <button
          className='modal__btn--cancel'
          onClick={props.stopDeleteDataSelected}
        >
          Cancel
        </button>
      </Modal>
    </div>
  );
};

export default ConfirmationModal;
