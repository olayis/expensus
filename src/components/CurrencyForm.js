import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';

export default class CurrencyForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      currency: props.currency,
      currencyError: '',
    };
  }

  onCurrencyChange = (e) => {
    const currency = e.target.value;
    this.setState(() => ({ currency }));
  };

  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.currency) {
      this.setState(() => ({
        currencyError: 'Please select a valid currency',
      }));
    } else {
      this.setState(() => ({ currencyError: '' }));
      this.props.onSubmit({
        currency: this.state.currency.trim(),
      });
    }
  };

  render() {
    return (
      <form className='input-field' onSubmit={this.onSubmit}>
        <div className='input-field__wrapper'>
          <select
            className='dropdown dropdown--currency'
            value={this.state.currency}
            onChange={this.onCurrencyChange}
            aria-label='Select Currency'
            name='currency'
          >
            <option value=''>-- Select Currency --</option>
            <option value='Baht'>Baht (฿)</option>
            <option value='Danish Krone'>Danish Krone (DKK)</option>
            <option value='Dollar'>Dollar ($)</option>
            <option value='Dong'>Dong (₫)</option>
            <option value='Euro'>Euro (€)</option>
            <option value='Forint'>Forint (Ft)</option>
            <option value='Hryvnia'>Hryvnia (₴)</option>
            <option value='Koruna'>Koruna (Kč)</option>
            <option value='Lira'>Lira (₺)</option>
            <option value='Lev'>Lev (лв)</option>
            <option value='Naira'>Naira (₦)</option>
            <option value='Norwegian krone'>Norwegian krone (kr)</option>
            <option value='Pound Sterling'>Pound Sterling (£)</option>
            <option value='Rand'>Rand (R)</option>
            <option value='Real'>Real (R$)</option>
            <option value='Ruble'>Ruble (₽)</option>
            <option value='Swiss Franc'>Swiss Franc (CHF)</option>
            <option value='Yen'>Yen (¥)</option>
            <option value='Yuan'>Yuan (¥)</option>
            <option value='Zloty'>Złoty (PLN)</option>
          </select>
          {this.state.currencyError && (
            <span className='input-field__error-msg'>
              <FontAwesomeIcon className='fa-icon' icon={faExclamationCircle} />
              {this.state.currencyError}
            </span>
          )}
        </div>
        <button className='btn btn-currency'>Change Currency</button>
      </form>
    );
  }
}
