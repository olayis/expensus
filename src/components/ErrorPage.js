import React from 'react';
import { history } from '../routers/AppRouter';
import engineer from '../../public/img/QA-Engineer.svg';

const ErrorPage = () => (
  <div className='content-container'>
    <div className='error'>
      <img
        src={engineer}
        alt='A Quality Assurance Engineer'
        className='error__illustration'
        width='1075'
        height='585'
      />
      <h2 className='error__text'>Oops! Something went wrong.</h2>
      <div className='error__explanation'>
        Click here to try again.
        <span className='error__action'>
          <button className='btn btn--try-again' onClick={() => history.go(0)}>
            Try Again
          </button>
        </span>
      </div>
    </div>
  </div>
);

export default ErrorPage;
