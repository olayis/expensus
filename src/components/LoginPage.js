import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import TypeIt from 'typeit-react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faGoogle,
  faTwitter,
  faFacebookF,
  faYahoo,
} from '@fortawesome/free-brands-svg-icons';
import randomQuote from '../utils/budgetQuotes/quotes';
import expensus_logo from '../../public/img/expensus-blue.svg';
import {
  startGoogleLogin,
  startFacebookLogin,
  startTwitterLogin,
  startYahooLogin,
} from '../actions/auth';
import {
  cancelAuthNetworkError,
  cancelAuthOtherErrors,
} from '../actions/authErrors';
import Footer from './Footer';

const { quote, author } = randomQuote();

export const LoginPage = ({
  startGoogleLogin,
  startFacebookLogin,
  startTwitterLogin,
  startYahooLogin,
  authErrors,
  cancelAuthNetworkError,
  cancelAuthOtherErrors,
}) => (
  <>
    <div className='split-layout'>
      <div className='split-layout__left'>
        <div className='login-content'>
          <div className='login-content__logo-box'>
            <Link to='/'>
              <img
                src={expensus_logo}
                alt='Expensus Logo'
                className='login-content__logo'
                width='195'
                height='37'
              />
            </Link>
          </div>
          <h1 className='login-content__message'>
            Hi! Sign in to start managing your expenses.
          </h1>
          <button
            onClick={startGoogleLogin}
            className='btn btn__login btn__login--google'
          >
            <FontAwesomeIcon icon={faGoogle} /> <h2>Continue with Google</h2>
          </button>
          <button
            onClick={startFacebookLogin}
            className='btn btn__login btn__login--facebook'
          >
            <FontAwesomeIcon icon={faFacebookF} />{' '}
            <h2>Continue with Facebook</h2>
          </button>
          <button
            onClick={startTwitterLogin}
            className='btn btn__login btn__login--twitter'
          >
            <FontAwesomeIcon icon={faTwitter} /> <h2>Continue with Twitter</h2>
          </button>
          <button
            onClick={startYahooLogin}
            className='btn btn__login btn__login--yahoo'
          >
            <FontAwesomeIcon icon={faYahoo} /> <h2>Continue with Yahoo</h2>
          </button>
        </div>
        <Footer />
      </div>
      <div className='split-layout__right'>
        <div className='quotes'>
          <TypeIt>
            <span className='quotes--text'>â€œ{quote}"</span>
            <br />
            <span className='quotes--author'>&mdash; {author}</span>
          </TypeIt>
        </div>
      </div>
    </div>
    {authErrors.networkError ? (
      <div className='auth-error'>
        <button className='auth-error__close' onClick={cancelAuthNetworkError}>
          &#10006;
        </button>
        <span className='auth-error__text'>Network Error.</span>
        <span className='auth-error__explanation'>
          Please check your internet connection and try again.
        </span>
      </div>
    ) : (
      ''
    )}
    {authErrors.otherErrors ? (
      <div className='auth-error'>
        <button className='auth-error__close' onClick={cancelAuthOtherErrors}>
          &#10006;
        </button>
        <span className='auth-error__text'>
          Oops! Something went wrong while authenticating...
        </span>
        <span className='auth-error__explanation'>Please try again.</span>
      </div>
    ) : (
      ''
    )}
  </>
);

const mapStateToProps = (state) => {
  return {
    authErrors: state.authErrors,
  };
};

const mapDispatchToProps = (dispatch) => ({
  startGoogleLogin: () => dispatch(startGoogleLogin()),
  startFacebookLogin: () => dispatch(startFacebookLogin()),
  startTwitterLogin: () => dispatch(startTwitterLogin()),
  startYahooLogin: () => dispatch(startYahooLogin()),
  cancelAuthNetworkError: () => dispatch(cancelAuthNetworkError()),
  cancelAuthOtherErrors: () => dispatch(cancelAuthOtherErrors()),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
