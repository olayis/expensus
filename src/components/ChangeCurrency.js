import React from 'react';
import { connect } from 'react-redux';
import { startChangeLocale } from '../actions/locale';
import CurrencyForm from './CurrencyForm';

export class ChangeCurrency extends React.Component {
  onSubmit = ({ currency }) => {
    this.props.startChangeLocale({ currency });
    this.props.history.push('/');
    this.props.history.go(0);
  };

  goBack = () => {
    this.props.history.goBack();
  };

  render() {
    return (
      <main className='main-content'>
        <div className='content-container'>
          <div className='component-container component-container--currency'>
            <CurrencyForm
              onSubmit={this.onSubmit}
              currency={this.props.currency}
              goBack={this.goBack}
            />
          </div>
        </div>
      </main>
    );
  }
}

const mapStateToProps = (state) => ({
  currency: state.locale ? state.locale.currency : '',
});

const mapDispatchToProps = (dispatch) => ({
  startChangeLocale: (currency) => dispatch(startChangeLocale(currency)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChangeCurrency);
