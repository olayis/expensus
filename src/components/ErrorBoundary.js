import React from 'react';
import store from '../store/configureStore';
import ErrorPage from './ErrorPage';
import startLogError from '../actions/logError';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    process.env.NODE_ENV === 'Production'
      ? store.dispatch(startLogError(error, errorInfo))
      : '';
  }

  render() {
    if (this.state.hasError) {
      return <ErrorPage />;
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
