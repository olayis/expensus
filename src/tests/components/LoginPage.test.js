import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should call startGoogleLogin on button click', () => {
  const startGoogleLogin = jest.fn();
  const wrapper = shallow(<LoginPage startGoogleLogin={startGoogleLogin} />);
  wrapper.find('.btn__login--google').simulate('click');
  expect(startGoogleLogin).toHaveBeenCalled();
});

test('should call startFacebookLogin on button click', () => {
  const startFacebookLogin = jest.fn();
  const wrapper = shallow(
    <LoginPage startFacebookLogin={startFacebookLogin} />
  );
  wrapper.find('.btn__login--facebook').simulate('click');
  expect(startFacebookLogin).toHaveBeenCalled();
});

test('should call startTwitterLogin on button click', () => {
  const startTwitterLogin = jest.fn();
  const wrapper = shallow(<LoginPage startTwitterLogin={startTwitterLogin} />);
  wrapper.find('.btn__login--twitter').simulate('click');
  expect(startTwitterLogin).toHaveBeenCalled();
});

test('should call startYahooLogin on button click', () => {
  const startYahooLogin = jest.fn();
  const wrapper = shallow(<LoginPage startYahooLogin={startYahooLogin} />);
  wrapper.find('.btn__login--yahoo').simulate('click');
  expect(startYahooLogin).toHaveBeenCalled();
});
