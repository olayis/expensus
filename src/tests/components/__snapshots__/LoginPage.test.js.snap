// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page correctly 1`] = `
<div
  className="split-layout"
>
  <div
    className="split-layout__left"
  >
    <div
      className="login-content"
    >
      <div
        className="login-content__logo-box"
      >
        <Link
          to="/"
        >
          <img
            alt="Expensus Logo"
            className="login-content__logo"
            height="37"
            src="SvgrURL"
            width="195"
          />
        </Link>
      </div>
      <h1
        className="login-content__message"
      >
        Hi! Sign in to start managing your expenses.
      </h1>
      <button
        className="btn btn__login btn__login--google"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                488,
                512,
                Array [],
                "f1a0",
                "M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z",
              ],
              "iconName": "google",
              "prefix": "fab",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         Continue with Google
      </button>
      <button
        className="btn btn__login btn__login--facebook"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                320,
                512,
                Array [],
                "f39e",
                "M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z",
              ],
              "iconName": "facebook-f",
              "prefix": "fab",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         Continue with Facebook
      </button>
      <button
        className="btn btn__login btn__login--twitter"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f099",
                "M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z",
              ],
              "iconName": "twitter",
              "prefix": "fab",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         Continue with Twitter
      </button>
      <button
        className="btn btn__login btn__login--yahoo"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f19e",
                "M223.69,141.06,167,284.23,111,141.06H14.93L120.76,390.19,82.19,480h94.17L317.27,141.06Zm105.4,135.79a58.22,58.22,0,1,0,58.22,58.22A58.22,58.22,0,0,0,329.09,276.85ZM394.65,32l-93,223.47H406.44L499.07,32Z",
              ],
              "iconName": "yahoo",
              "prefix": "fab",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         Continue with Yahoo
      </button>
    </div>
    <LoginPageFooter />
  </div>
  <div
    className="split-layout__right"
  >
    <div
      className="quotes"
    >
      <U
        element="span"
        getAfterInit={[Function]}
        getBeforeInit={[Function]}
        options={Object {}}
      >
        <span
          className="quotes--text"
        >
          “
          A budget tells us what we can't afford, but it doesn't keep us from buying it.
          "
        </span>
        <br />
        <span
          className="quotes--author"
        >
          — 
          William Feather
        </span>
      </U>
    </div>
  </div>
</div>
`;
